lib/ex_machina/ecto.ex

```diff
@@ -66,6 +73,50 @@ defmodule ExMachina.Ecto do
     |> drop_ecto_fields
   end
 
+  @doc """
+  Same as `params_for/2`, but it inserts all belongs_to associations and sets the
+  foreign_keys.
+
+  ## Example
+
+      def article_factory do
+        %MyApp.Article{title: "An Awesome Article", author: build(:author)}
+      end
+
+      # Inserts an author and returns
+      # %{title: "An Awesome Article", author_id: 12}
+      params_with_assocs(:article)
+  """
+  def params_with_assocs(module, factory_name, attrs \\ %{}) do
+    module.build(factory_name, attrs)
+    |> insert_belongs_to_assocs(module)
+    |> drop_ecto_fields
+  end
+
+  defp insert_belongs_to_assocs(record = %{__struct__: struct, __meta__: %{__struct__: Ecto.Schema.Metadata}}, module) do
+    Enum.reduce(struct.__schema__(:associations), record, fn(association_name, record) ->
+      case struct.__schema__(:association, association_name) do
+        association = %{__struct__: Ecto.Association.BelongsTo} ->
+          insert_built_belongs_to_assoc(
+            module,
+            association.owner_key,
+            association_name,
+            record
+          )
+        _ -> record
+      end
+    end)
+  end
+
+  def insert_built_belongs_to_assoc(module, foreign_key, association_name, record) do
```

Comment:
And maybe that should also be changed for the other functions. Might be better for a separate PR though. I'd be fine with this as-is